packages/menu/src/Menu.tsx

```diff
@@ -88,46 +89,152 @@ function Menu({
+        if (ref == null) {
+          return;
+        }
+
+        refs.push(ref);
+        currentChildRef = ref;
+
+        if (open && hasSetInitialFocus.current === false) {
+          setFocus(refs[0]);
+          // This allows the focus state to be shown via the LeafyGreen Provider
+          setUsingKeyboard(true);
+          hasSetInitialFocus.current = true;
+        }
+      };
+
+      const onFocus = ({ target }: { target: HTMLElement }) => {
+        setFocused(target);
+      };
+
+      if (isComponentType<'SubMenu'>(child, 'SubMenu')) {
+        if (titleArr.includes(child.props.title)) {
+          throw new Error('SubMenu titles must be unique');
+        }
+
+        titleArr.push(child.props.title);
+
+        if (child.props.active && !hasSetInitialOpen.current) {
+          setCurrentSubMenu(child);
+          hasSetInitialOpen.current = true;
+        }
+
+        const isCurrentSubMenu =
+          currentSubMenu?.props.title === child?.props.title;
+
+        return React.cloneElement(child, {
+          ref: setRef,
+          open: isCurrentSubMenu,
+          setOpen: (state: boolean) => {
+            if (currentChildRef) {
+              setFocused(currentChildRef);
+            }
+
+            setCurrentSubMenu(state ? child : null);
+          },
+          onKeyDown: (e: KeyboardEvent) => {
+            if (e.keyCode === keyMap.ArrowLeft && isCurrentSubMenu) {
+              setCurrentSubMenu(null);
+            }
+
+            if (e.keyCode === keyMap.ArrowRight) {
+              setCurrentSubMenu(child);
+            }
+          },
+          onFocus,
+          children: updateChildren(child.props.children),
+          onExited: () => {
+            setClosed(curr => !curr);
+          },
+        });
+      }
+
+      if (
+        isComponentType<'MenuItem'>(child, 'MenuItem') ||
+        isComponentType<'FocusableMenuItem'>(child, 'FocusableMenuItem')
+      ) {
+        return React.cloneElement(child, {
+          ref: setRef,
+          onFocus,
+        });
+      }
+
+      if (child.props?.children) {
+        return React.cloneElement(child, {
+          children: updateChildren(child.props.children),
+        });
+      }
+
+      return child;
+    });
+  }
 
   const [focused, setFocused] = useState<HTMLElement>(refs[0] || null);
-  const isControlled = typeof controlledOpen === 'boolean';
   const [uncontrolledOpen, uncontrolledSetOpen] = useState(false);
-  const open = isControlled ? controlledOpen : uncontrolledOpen;
-  const setOpen = isControlled ? controlledSetOpen : uncontrolledSetOpen;
+  const setOpen =
+    (typeof controlledOpen === 'boolean' && controlledSetOpen) ||
+    uncontrolledSetOpen;
+  const open = controlledOpen ?? uncontrolledOpen;
+
+  // When a SubMenu becomes open, it's set to currentSubMenu, and we focus on the first child.
+  useEffect(() => {
+    if (!currentSubMenu) {
+      const focusedRefIndex = refs.indexOf(focused);
```

Comment:
Rather than using optional chaining to check for the existence of `subMenu` before focusing, it might be better to instead check that focusedRefIndex is not `-1`.

Reply:
Ditto in the `else` block.

Reply:
👍